sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,838 @@
+     *        be applied. If omitted or greater than 2,000, the request will process up to 2,000
+     *        items.
+     */
+    int32_t MaxRecords = int32_t();
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories. The value is a
+     *        comma-separated list of access control entries. Each access control entry (ACE)
+     *        consists of a scope, a type, a user or group identifier, and permissions in the format
+     *        "[scope:][type]:[id]:[permissions]".
+     */
+    std::string Acl;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::SetProperties
+   */
+  struct SetPathPropertiesOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief User-defined metadata to be stored with the filesystem.
+     *        Note that the string may only contain ASCII characters in the
+     *        ISO-8859-1 character set.
+     */
+    std::map<std::string, std::string> Metadata;
+
+    /**
+     * @brief Sets the blob's cache control. If specified, this property is stored with the
+     *        blob and returned with a read request.
+     */
+    std::string CacheControl;
+
+    /**
+     * @brief Sets the blob's content type. If specified, this property is stored
+     *        with the blob and returned with a read request.
+     */
+    std::string ContentType;
+
+    /**
+     * @brief Sets the blob's Content-Disposition header.
+     */
+    std::string ContentDisposition;
+
+    /**
+     * @brief Sets the blob's content encoding. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentEncoding;
+
+    /**
+     * @brief Set the blob's content language. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentLanguage;
+
+    /**
+     * @brief Specify an ETag value to operate only on path with a matching value.
+     */
+    std::string IfMatch;
+
+    /**
+     * @brief Specify an ETag value to operate only on path without a matching value.
+     */
+    std::string IfNoneMatch;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::Create
+   * @remark Some optional parameter is mandatory in certain combination.
+   *         More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create
+   */
+  struct PathCreateOptions : public SharedRequestOptions
```

Comment:
Since you're having that `CreateFileSystemOptions`, would it be better if this is named `CreatePathOptions`?

Reply:
ditto for delete, getproperties etc.

Reply:
This is to avoid a name clash with protocol API.

Reply:
I think protocol layer options are nested in class of each resource. e.g. `CreatePathOptions` are defined in `class Path`?

Reply:
https://github.com/Azure/azure-sdk-for-cpp/blob/11f2b8255dab0467f6de944d7aff6da804b833e0/sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp#L1417

Reply:
That's actually true. When I wrote this code, there was a name clash. I didn't realize that this is no longer the case. I'll rename those options.

Reply:
Actually I'll rename the file system ones since they are the ones not being consistent.

Reply:
fine, as long as they are consistent